dask_kubernetes/operator/daskcluster.py

```diff
@@ -191,14 +196,85 @@ async def daskworkergroup_create(spec, name, namespace, logger, **kwargs):
+        name,
+        spec.get("image"),
+        spec.get("replicas"),
+        spec.get("resources"),
+        spec.get("env"),
+    )
+    kopf.adopt(data)
+    num_workers = spec["replicas"]
     for i in range(1, num_workers + 1):
-        data = build_worker_pod_spec(scheduler_name, namespace, spec.get("image"), i)
+        data = build_worker_pod_spec(
+            name, namespace, spec.get("image"), i, scheduler_name
+        )
         kopf.adopt(data)
         worker_pod = api.create_namespaced_pod(
             namespace=namespace,
             body=data,
         )
-    # await wait_for_scheduler(name, namespace)
     logger.info(f"{num_workers} Worker pods in created in {namespace}")
+
+
+@kopf.on.update("daskworkergroup")
+async def daskworkergroup_update(spec, name, namespace, logger, **kwargs):
+    api = kubernetes.client.CoreV1Api()
+
+    scheduler = kubernetes.client.CustomObjectsApi().list_cluster_custom_object(
+        group="kubernetes.dask.org", version="v1", plural="daskclusters"
+    )
+    scheduler_name = scheduler["items"][0]["metadata"]["name"]
+    workers = api.list_namespaced_pod(
+        namespace=namespace,
+        label_selector=f"dask.org/workergroup-name={name}",
+    )
+    current_workers = len(workers.items)
+    desired_workers = spec["replicas"]
+    workers_needed = desired_workers - current_workers
+    if workers_needed > 0:
+        for i in range(current_workers + 1, desired_workers + 1):
+            data = build_worker_pod_spec(
+                name, namespace, spec.get("image"), i, scheduler_name
+            )
+            kopf.adopt(data)
+            worker_pod = api.create_namespaced_pod(
+                namespace=namespace,
+                body=data,
+            )
+        logger.info(f"Scaled worker group {name} up to {spec['replicas']} workers.")
+    if workers_needed < 0:
+        for i in range(current_workers, desired_workers, -1):
+            worker_pod = api.delete_namespaced_pod(
+                name=f"{name}-worker-{i}",
+                namespace=namespace,
+            )
+        logger.info(f"Scaled worker group {name} down to {spec['replicas']} workers.")
+
+
+@kopf.on.delete("daskcluster")
+async def daskcluster_delete(spec, name, namespace, logger, **kwargs):
+    api = kubernetes.client.CustomObjectsApi()
+    workergroups = api.list_cluster_custom_object(
+        group="kubernetes.dask.org", version="v1", plural="daskworkergroups"
+    )
+    workergroups = api.delete_collection_namespaced_custom_object(
+        group="kubernetes.dask.org",
+        version="v1",
+        plural="daskworkergroups",
+        namespace=namespace,
+    )
+
+
+@kopf.on.delete("daskworkergroup")
+async def daskworkergroup_delete(spec, name, namespace, logger, **kwargs):
+    api = kubernetes.client.CoreV1Api()
+    workers = api.delete_collection_namespaced_pod(
+        namespace=namespace,
+        label_selector=f"dask.org/workergroup-name={name}",
+    )
```

Comment:
I don't think we need this. When we use `kopf.adopt` that ties the resources together and when we create the top level `DaskCluster` Kubernetes cleans everything up for us.